common/v2/components/AccountList.tsx

```diff
@@ -214,22 +285,70 @@ function buildAccountTable(
   overlayRows?: number[],
   setDeletingIndex?: any
 ) {
+  const [sortingState, setSortingState] = useState(initialSortingState);
   const { totalFiat } = useContext(StoreContext);
   const { getAssetRate } = useContext(RatesContext);
   const { settings } = useContext(SettingsContext);
   const { addressBook, updateAddressBooks, createAddressBooks } = useContext(AddressBookContext);
+
+  const updateSortingState = (id: IColumnValues) => {
+    const currentBtnState = sortingState.sortState[id];
+    if (currentBtnState.indexOf('-reverse') > -1) {
+      const newActiveSort = currentBtnState.split('-reverse')[0] as ISortTypes;
+      setSortingState({
+        sortState: {
+          ...sortingState.sortState,
+          [id]: newActiveSort
+        },
+        activeSort: newActiveSort
+      });
+    } else {
+      const newActiveSort = (currentBtnState + '-reverse') as ISortTypes;
+      setSortingState({
+        sortState: {
+          ...sortingState.sortState,
+          [id]: newActiveSort
+        },
+        activeSort: newActiveSort
+      });
+    }
+  };
+
+  const getColumnSortDirection = (id: IColumnValues): boolean =>
+    sortingState.sortState[id].indexOf('-reverse') > -1 ? true : false;
+
+  const convertColumnToClickable = (id: IColumnValues) => (
+    <div onClick={() => updateSortingState(id)}>
+      {translateRaw(id)} <IconArrow isFlipped={getColumnSortDirection(id)} />
+    </div>
+  );
+
   const columns = [
-    translateRaw('ACCOUNT_LIST_LABEL'),
-    translateRaw('ACCOUNT_LIST_ADDRESS'),
-    translateRaw('ACCOUNT_LIST_NETWORK'),
-    <HeaderAlignment key={'ACCOUNT_LIST_VALUE'} align="center">
+    convertColumnToClickable('ACCOUNT_LIST_LABEL'),
+    convertColumnToClickable('ACCOUNT_LIST_ADDRESS'),
+    convertColumnToClickable('ACCOUNT_LIST_NETWORK'),
+    <HeaderAlignment
+      key={'ACCOUNT_LIST_VALUE'}
+      align="center"
+      onClick={() => updateSortingState('ACCOUNT_LIST_VALUE')}
+    >
       {translateRaw('ACCOUNT_LIST_VALUE')}
+      <IconArrow isFlipped={getColumnSortDirection('ACCOUNT_LIST_VALUE')} />
     </HeaderAlignment>,
     <HeaderAlignment key={'ACCOUNT_LIST_DELETE'} align="center">
       {translateRaw('ACCOUNT_LIST_DELETE')}
     </HeaderAlignment>
   ];
 
+  const getFullTableData = accounts
+    .map((account, index) => {
+      const addressCard: ExtendedAddressBook | undefined = getLabelByAccount(account, addressBook);
```

Comment:
How about including the `ExtendedAddressBook` directly inside `StoreAccount.label` that way the component wouldn't need to know about our data structure.

Reply:
Lets do this in a different pr when we go to change StoreAccount c: